src/uptane/uptanerepository.cc

```diff
@@ -42,13 +42,22 @@ void Repository::putManifest() { putManifest(Json::nullValue); }
 void Repository::putManifest(const Json::Value &custom) {
   Json::Value version_manifest;
   version_manifest["primary_ecu_serial"] = config.uptane.primary_ecu_serial;
-  version_manifest["ecu_version_manifest"] = transport.getManifests();
+
+  Json::Value secondary_manifests = transport.getManifests();
+  if (!secondary_manifests.empty()) {
+    version_manifest["ecu_version_manifest"] = transport.getManifests();
+    for (Json::Value::iterator it = secondary_manifests.begin(); it != secondary_manifests.end(); ++it) {
+      version_manifest["ecu_version_manifests"][(*it)["signed"]["ecu_serial"].asString()] = *it;
+    }
+  }
 
   Json::Value unsigned_ecu_version =
       OstreePackage::getEcu(config.uptane.primary_ecu_serial, config.ostree.sysroot).toEcuVersion(custom);
   Json::Value ecu_version_signed =
       Crypto::signTuf(config.uptane.private_key_path, config.uptane.public_key_path, unsigned_ecu_version);
   version_manifest["ecu_version_manifest"].append(ecu_version_signed);
+  version_manifest["ecu_version_manifests"][config.uptane.primary_ecu_serial] = ecu_version_signed;
+
   Json::Value tuf_signed =
```

Comment:
I get the following error from the production server with this change:

[2017-07-31 11:43:49.718849]: response: {"code":"signature_not_valid","description":"The given signature is not valid"}